sansa-inference-spark/src/main/scala/net/sansa_stack/inference/spark/forwardchaining/axioms/TransitiveReasoner.scala

```diff
@@ -0,0 +1,102 @@
+package net.sansa_stack.inference.spark.forwardchaining.axioms
+
+import org.apache.spark.rdd.RDD
+import org.semanticweb.owlapi.apibinding.OWLManager
+import org.semanticweb.owlapi.model._
+
+import scala.reflect.ClassTag
+
+/*
+ *  A class for computing Transitive Closure for a set of given OWLAxioms with different formats.
+ */
+
+class TransitiveReasoner extends Serializable{
```

Comment:
Some issues I see here:

- The transitive reasoner class name promises more than it actually delivers. Right now it does not consider further transitive properties besides `rdfs:subClassof` and `rdfs:subPropertyOf`. To get transitive properties one would have to filter for `OWLTransitiveObjectPropertyAxiom` objects.
- I'm missing a clear interface for the transitive reasoner class. Right now one can call any of the public methods but would have to do the filtering in advance. In my opinion it would be cleaner to have one main public method which gets the whole RDD and does the filtering for the known 'transitive axioms' (`OWLSubClassOfAxiom`, `OWLSubObjectPropertyOfAxiom`, `OWLSubDataPropertyOfAxiom`, `OWLSubAnnotationPropertyOfAxiom`) and those that make use of further transitive properties.

Reply:
* Filtering for _OWLTransitiveObjectPropertyAxiom_ will come in the implementation of the OWL Horst rules, I implement only the RDFs rules.
* I need the axioms to be filtered before sending them becuase I make use of the filtered RDDs in the implementations of some rules.
* I made some changes in the _TransitiveReasoner_